src/elements/content-sidebar/versions/VersionsItemActions.js

```diff
@@ -0,0 +1,110 @@
+import PlainButton from '../../../components/plain-button';
+import { Menu, MenuItem } from '../../../components/menu';
+import './VersionsItemActions.scss';
+
+type Props = {
+    isCurrent: boolean,
+    isDeleted: boolean,
+    onDelete?: () => void,
+    onDownload?: () => void,
+    onPreview?: () => void,
+    onPromote?: () => void,
+    onRestore?: () => void,
+    permissions: BoxItemPermission,
+};
+
+const ICON_SIZE = { height: 12, width: 12 };
+
+const handleMenuClose = (event: SyntheticEvent<>) => {
+    event.stopPropagation();
+};
+
+const handleToggleClick = (event: SyntheticMouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation();
+};
+
+const VersionsItemActions = ({
+    isCurrent = false,
+    isDeleted = false,
+    onDelete,
+    onDownload,
+    onPreview,
+    onPromote,
+    onRestore,
+    permissions,
+}: Props) => {
+    const { can_delete, can_download, can_preview, can_upload } = permissions;
+    const showDelete = can_delete && !isDeleted && !isCurrent;
+    const showDownload = can_download && !isDeleted;
+    const showPreview = can_preview && !isDeleted;
+    const showPromote = can_upload && !isDeleted && !isCurrent;
+    const showRestore = can_delete && isDeleted;
+
+    if (!showDelete && !showDownload && !showPreview && !showPromote && !showRestore) {
+        return null;
+    }
+
+    return (
+        <DropdownMenu
+            className="bcs-VersionsItemActions"
+            constrainToScrollParent
+            isRightAligned
+            onMenuClose={handleMenuClose}
+        >
+            <PlainButton className="bcs-VersionsItemActions-toggle" onClick={handleToggleClick}>
+                <IconEllipsis height={4} width={14} title={<FormattedMessage {...messages.versionActionToggle} />} />
+            </PlainButton>
+            <Menu>
+                {showPreview && (
+                    <MenuItem className="bcs-VersionsItemActions-item" onClick={onPreview}>
+                        <IconOpenWith {...ICON_SIZE} />
+                        <FormattedMessage {...messages.versionActionPreview} />
+                    </MenuItem>
+                )}
+
+                {showDownload && (
+                    <MenuItem className="bcs-VersionsItemActions-item" onClick={onDownload}>
+                        <IconDownload {...ICON_SIZE} />
+                        <FormattedMessage {...messages.versionActionDownload} />
+                    </MenuItem>
+                )}
+                {showPromote && (
+                    <MenuItem className="bcs-VersionsItemActions-item" onClick={onPromote}>
+                        <IconUpload {...ICON_SIZE} />
+                        <FormattedMessage {...messages.versionActionPromote} />
+                    </MenuItem>
+                )}
+                {showRestore && (
+                    <MenuItem className="bcs-VersionsItemActions-item" onClick={onRestore}>
+                        <IconClockPast height={14} width={14} />
```

Comment:
Does this icon need to be tweaked to be consistent with the others? (Not of the scope of this PR but just wondering)

Reply:
Ideally, all our icons would have the same default dimensions, padding, and viewbox, but that's not the case currently. I know @tjgupta has put some effort into this area, but my understanding is that it's a pretty large effort from both us and design.